tefla/core/eval_metrices.py

```diff
@@ -0,0 +1,772 @@
+                        false_pos[i], true_neg[i])
+            elif element in ['false discovery rate', 'fdr']:
+                for i, cls in enumerate(classes):
+                    self.eval_result[cls]['fdr'] = calc_fdr(
+                        true_pos[i], false_pos[i])
+            elif element in ['false omission rate', 'for']:
+                for i, cls in enumerate(classes):
+                    self.eval_result[cls]['for'] = calc_for(
+                        false_neg[i], true_neg[i])
+            elif element in ['matthews correlatin coefficient', 'mcc']:
+                for i, cls in enumerate(classes):
+                    self.eval_result[cls]['mcc'] = calc_mcc(
+                        true_pos[i], true_neg[i], false_pos[i], false_neg[i])
+            elif element in ['kappa']:
+                for i, cls in enumerate(classes):
+                    self.eval_result[cls]['kappa'] = calc_kappa(
+                        self.truth, self.pred_max)
+            else:
+                raise Exception("invalid Evaluation Term")
+
+        if plot_list:
+            self.eval_plot_classification(
+                plot_list, classes, true_pos, false_pos, false_neg, true_neg)
+        if over_all:
+            self.calc_overall()
+        return self.eval_result
+
+    def eval_regression(
+            self,
+            truth_file,
+            pred_files,
+            eval_list,
+            plot_list,
+            ensemble_weights=None):
+        """
+            This function calculates the evaluation matrices
+            required by the user for regression problems.
+        Args:
+            truth_file: A csv file containing ids and truth annotations.
+            pred_files: A list of csv files containing ids and prediction annotations.
+            eval_list: A list of evaluation measures.
+            plot_list: A list of evaluation plots.
+            ensemble_weights: Weights for each class in case of ensemble, default None.
+        Returns:
+            A dictionary containing evaluation result.
+        Raises: Invalid evaluation term if a term is not present in supported list.
+        """
+        self.eval_result = {}
+        self.read_data(truth_file, pred_files)
+        eval_list = [element.strip().lower()
+                     for element in eval_list]
+        if self.ensemble:
+            ensemble_voting = "soft"
+            self.eval_ensemble(ensemble_voting, ensemble_weights)
+        for element in eval_list:
+            if element in ['mae', 'mean absolute error']:
+                self.eval_result['mae'] = calc_mae(self.truth, self.pred)
+            elif element in ['mse', 'mean squared error']:
+                self.eval_result['mse'] = calc_mse(self.truth, self.pred)
+            elif element in ['rmse', 'root mean squared error']:
+                self.eval_result['rmse'] = calc_rmse(self.truth, self.pred)
+            else:
+                raise Exception("invalid Evaluation Term")
```

Comment:
rather than using a general exception use the specific exception here `raise ValueError("Invalid evaluation term")`

Reply:
updated